docs/rfcs/PipelineComposition.md

```diff
@@ -0,0 +1,73 @@
+# RFC: High Level MLIR Pass Pipeline Composition
+
+Ivan Butygin, Renato Golin
+
+## Motivation
+
+TBD use cases from IREE, TPP
```

Comment:
It would help reader to start with a motivation. I assume that the dependency-based graph would avoid some mistake when user configure the pipeline manually and unintentionally break the dependency. Is it correct?

Reply:
Expanded motivation.